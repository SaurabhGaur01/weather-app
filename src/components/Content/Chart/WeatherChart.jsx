import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BarChart from 'react-bar-chart';
   
const margin = {top: 20, right: 20, bottom: 30, left: 40};
   
class WeatherChart extends React.Component{
    constructor(props) {
      super(props);
      let width = window.innerWidth;
      if (width > 768) {
        this.state = {
          width: 500
        }
      }else{
        this.state = {
          width: 300
        }
      } 
    }
 
    render() {
      const { chartData } = this.props;
      return (
          <div ref='root'>
            <BarChart 
              width={this.state.width}
              height={200}
              margin={margin}
              data={chartData}
            />
          </div>
      );
    }
};

WeatherChart.propTypes = {
  chartData: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
  })),
};

const mapStateToProps = state => ({
  chartData: state.chartData,
});

export { WeatherChart as TestableWeatherChart };

export default connect(mapStateToProps)(WeatherChart);