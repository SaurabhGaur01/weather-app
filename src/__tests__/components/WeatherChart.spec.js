import React from 'react';
import { shallow } from 'enzyme';
import { TestableWeatherChart } from '../../components/Content/Chart/WeatherChart';

describe('WeatherChart Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            chartData: [
                {
                    text: '03:00',
                    value: 50.
                },
                {
                    text: '04:00',
                    value: 60.
                },
            ]
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableWeatherChart {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
})