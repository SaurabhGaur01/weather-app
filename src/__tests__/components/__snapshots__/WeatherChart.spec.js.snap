// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherChart Component should render correctly 1`] = `
<div>
  <BarChart
    data={
      Array [
        Object {
          "text": "03:00",
          "value": 50,
        },
        Object {
          "text": "04:00",
          "value": 60,
        },
      ]
    }
    height={200}
    margin={
      Object {
        "bottom": 30,
        "left": 40,
        "right": 20,
        "top": 20,
      }
    }
    width={500}
  />
</div>
`;
